### PR TITLE
Add org.yamagi.YamagiQ2

### PR DESCRIPTION
**Pull request by Eonfge**
Related to #1160

This was a very easy one. High quality build scripts and code, which compiled quite fast. It's fully FHS compatible so a simple `persist=.yq2` was all that was required to provide the application with all proprietary information.

Biggest outstanding question to you, what to do with the icon? Yamagi uses the official Quake 2 icon (https://github.com/yquake2/yquake2/issues/106) but I'm not sure what the policy is on icons of existing Trademarks.

### CODE REVIEW

File: org.yamagi.YamagiQ2.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      <li>In comparison with the original client over 1.500 bugs were fixed. The Yamagi Quake II client is very well tested and stable. The dedicated server is able to run for weeks without a single restart. Last but not least the code is fully 64 bit clean and highly portable.</li>
+    </ul>
+    <p>
+     Commercial data files are required to run the supported game. These can be aquired though a multitude of sources. See the manual for more info on this.
+    </p>
+  </description>
+
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/flathub/org.yamagi.YamagiQ2/master/images/image_01.png</image>
```
Comment (bilelmoussaoui) [inline org.yamagi.YamagiQ2.appdata.xml]: Use your own repository for screenshots & switch back to this once the repo is created.

Comment (Eonfge) [reply org.yamagi.YamagiQ2.appdata.xml]: Updated.

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by Eonfge** [comment]
@bilelmoussaoui Could you have a look at the builderbot? For this issue and for #1173, it seems to be stuck.

**Comment by bilelmoussaoui** [comment]
I'm getting this and nothing works
```
s: cannot access '/home/bilelmussaoui/.yq2/baseq2/pak0.pak': No such file or directory

Yamagi Quake II v7.41
=====================
Client build options:
 + OpenAL audio
Platform: Linux
Architecture: x86_64
Byte ordering: little endian

No Ogg Vorbis music tracks have been found, so there will be no music.
Using '/home/bilelmussaoui/.yq2//baseq2' for writing.
couldn't exec default.cfg
couldn't exec yq2.cfg
couldn't exec config.cfg
couldn't exec autoexec.cfg
Console initialized.

------- sound initialization -------
LoadLibrary(libopenal.so.1)
...opening OpenAL device:ok
...creating OpenAL context: ok
...making context current: ok

OpenAL settings:
AL_VENDOR: OpenAL Community
AL_RENDERER: OpenAL Soft
AL_VERSION: 1.1 ALSOFT 1.19.1
AL_EXTENSIONS: AL_EXT_ALAW AL_EXT_BFORMAT AL_EXT_DOUBLE AL_EXT_EXPONENT_DISTANCE AL_EXT_FLOAT32 AL_EXT_IMA4 AL_EXT_LINEAR_DISTANCE AL_EXT_MCFORMATS AL_EXT_MULAW AL_EXT_MULAW_BFORMAT AL_EXT_MULAW_MCFORMATS AL_EXT_OFFSET AL_EXT_source_distance_model AL_EXT_SOURCE_RADIUS AL_EXT_STEREO_ANGLES AL_LOKI_quadriphonic AL_SOFT_block_alignment AL_SOFT_deferred_updates AL_SOFT_direct_channels AL_SOFTX_events AL_SOFTX_filter_gain_ex AL_SOFT_gain_clamp_ex AL_SOFT_loop_points AL_SOFTX_map_buffer AL_SOFT_MSADPCM AL_SOFT_source_latency AL_SOFT_source_length AL_SOFT_source_resampler AL_SOFT_source_spatialize

Available OpenAL devices:
- Built-in Audio Analog Stereo

Current OpenAL device:
- OpenAL Soft

Number of OpenAL sources: 31

Sound sampling rate: 0
------------------------------------

SDL version is: 2.0.10
SDL video driver is "wayland".
----- refresher initialization -----
LoadLibrary(ref_gl1.so)
Byte ordering: little endian

Refresh: Yamagi Quake II OpenGL Refresher
Client: 7.41

Shutting down OpenAL.
Cmd_RemoveCommand: force_centerview not added
Cmd_RemoveCommand: +mlook not added
Cmd_RemoveCommand: -mlook not added
Shutting down input.
Cmd_RemoveCommand: modellist not added
Cmd_RemoveCommand: screenshot not added
Cmd_RemoveCommand: imagelist not added
Cmd_RemoveCommand: gl_strings not added
recursive shutdown
Error: Couldn't load pics/colormap.pcx
```

**Comment by Eonfge** [comment]
@bilelmoussaoui  Did you add the commercial, proprietary data?
Screenshot from 2019-09-25 12-08-00

It's supposed to give a Zenity error message when it can't find the required commercial data.

**Comment by barthalion** [comment]
I don't think we can use the icon unless it's part of Quake II source code.

Have you contacted upstream team if they would like to (co-)maintain their application on Flathub?

**Comment by Eonfge** [comment]
@barthalion I wanted to hear your opinion first. I created an issue on their board concerning the trademark: https://github.com/yquake2/yquake2/issues/462

**Comment by barthalion** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by barthalion** [comment]
LGTM (at least to the point zenity asks me for data). Good to merge when icon is replaced.

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

@barthalion I forgot this one, never realised that I never distributed it. It still works locally, so that's why.

Anyway, Upstream discussed the logo-issue and they told me that a (low-resolution) version of the logo was part of the GPL release. Debian considers that 'good enough' for their rigid FLOSS and trademarking standards.

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by Eonfge** [comment]
@barthalion Also an idea why the app icon fails?

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by Eonfge** [comment]
@barthalion Could you check why this suddenly passes? 

It seems that the image convert script as mentioned in your blogpost doesn't work:
https://discourse.flathub.org/t/a-128x128-icon-is-required-now/386/2

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by bilelmoussaoui** [comment]
bot, build org.yamagi.YamagiQ2

**Comment by Eonfge** [comment]
bot, build org.yamagi.YamagiQ2

@bilelmoussaoui It seems that Bert is not available. Could you give me a comment on the updated logo status? I've neglected this pull request a bit but I would like to finish it

**Comment by bilelmoussaoui** [comment]
> bot, build org.yamagi.YamagiQ2
> 
> @bilelmoussaoui It seems that Bert is not available. Could you give me a comment on the updated logo status? I've neglected this pull request a bit but I would like to finish it

I'm not sure what you mean by the logo status?

**Comment by Eonfge** [comment]
@bilelmoussaoui 

https://github.com/flathub/flathub/pull/1174#issuecomment-535411413

Followed by

https://github.com/flathub/flathub/pull/1174#issuecomment-617565700

**Comment by Eonfge** [comment]
Both @barthalion and @bilelmoussaoui, perhaps it's good if we go through this issue once more.

Originally, I packaged Yamagi Quake 2 and everything seemed to work. There is a proprietary component required, but a Zenity box will inform users about. Once the Id assets are in place, everything runs smooth.

When I proposed this issue, I did have a question concerning the trademark since It's an actual copy of the Quake logo. Then, you told me that I had to get that cleared before this package could be added to Flathub.

Well, I went upstream and I aksed them about it. And then the pull request slipped to the back of my mind. The package still runs on my machine, and I even made a small update, but I didn't come back to it.

https://github.com/yquake2/yquake2/issues/462

Upstream told me two things: 
1. There is a low-quality logo of Quake 2 included in the GPL release of the code
2. That was enough for Debian to have it approved for [contrib](https://www.debian.org/doc/debian-policy/ch-archive.html#the-contrib-archive-area)

So, when I pushed some updated to this repository a few weeks ago, this package came back into my focus, and now I wish to round it up. I hope that the trademarking issue is properly resolved and that this package can be added to Flathub. it's not a major package, but it's nice to finish it.

**Pull request closed without merging**